core/src/mindustry/logic/LExecutor.java

```diff
@@ -1363,13 +1363,21 @@ public void run(LExecutor exec){
             TeamData data = t.data();
 
             switch(type){
-                case unit -> exec.setobj(result, i < 0 || i >= data.units.size ? null : data.units.get(i));
+                case unit -> {
+                    UnitType type = exec.obj(extra) instanceof UnitType u ? u : null;
+                    if(type == null){
+                        exec.setobj(result, i < 0 || i >= data.units.size ? null : data.units.get(i));
+                    }else{
+                        var units = data.unitCache(type);
```

Comment:
`unitCache` can return null. This can crash.

Reply:
What about the `i < 0 || i >= data.units.size ? null : data.units.get(i)` below?

Reply:
I thought `unitCache` was `@Nullable`

Reply:
Fixed the `unitCount` nullpointerexception crash when there's no specific unit in extra